native/src/main/java/io/ballerina/stdlib/kafka/utils/KafkaUtils.java

```diff
@@ -94,6 +97,13 @@ private KafkaUtils() {
             caller.addNativeData(NATIVE_CONSUMER, consumer);
             caller.addNativeData(NATIVE_CONSUMER_CONFIG, consumerProperties);
             return new Object[]{caller, true, consumerRecordsArray, true};
+        } else if (parameterTypes.length == 1 && parameterTypes[0].getTag() == ARRAY_TAG) {
+            for (Object record : records) {
+                BMap<BString, Object> consumerRecord = populateConsumerRecord(
+                        (ConsumerRecord) record, keyType, valueType);
+                consumerRecordsArray.append(consumerRecord);
+            }
```

Comment:
We can get rid of these repeated lines by creating the `consumerRecord` outside the `if` at line 86 as anyway it is mandatory to have `kafka:ConsumerRecord[] records` as a parameter of the `onConsumerRecord` function.
Then check if there is an `OBJECT_TYPE_TAG` tag and if it is there create the `BObject caller`

Reply:
Fixed in the latest commits